### PR TITLE
Introduce Sorbet TypeChecking

### PR DESCRIPTION
**Pull request by tsmartt**

### CODE REVIEW

**Review by porteron — APPROVED** [review]

**Review by thypon — CHANGES_REQUESTED** [review]
- Can we exclude `rbi` from the changes? They seems autogenerated and we should not edit them anyway
- Can we split all the `# type: ignore` changes in a separated commit?

File: Gemfile [hunk 1]
```diff
@@ -136,6 +136,9 @@ gem "sidekiq-scheduler", "~> 3.0.1"
 # slim for view templates
 gem "slim-rails", "~> 3.1"
 
+gem "sorbet", group: :development
```
[the remark this answers is not in the document]
Comment (tsmartt) [reply Gemfile:139]: Done.

File: Gemfile [hunk 2]
```diff
@@ -146,6 +149,8 @@ gem "rotp", "~> 6.2.0"
 
 gem "webpacker", "~> 4.0.7"
 
+gem "sorbet-rails"
```
[the remark this answers is not in the document]
Comment (tsmartt) [reply Gemfile:152]: Done.

File: Gemfile [hunk 3]
```diff
@@ -178,6 +183,8 @@ group :development do
 
   # i18n-tasks helps you find and manage missing and unused translations.
   gem "i18n-tasks", "~> 0.9.12"
+
+  gem "tapioca", require: false
```
[the remark this answers is not in the document]
Comment (tsmartt) [reply Gemfile:187]: Done.

File: Gemfile [hunk 4]
```diff
@@ -136,6 +136,9 @@ gem "sidekiq-scheduler", "~> 3.0.1"
 # slim for view templates
 gem "slim-rails", "~> 3.1"
 
+gem "sorbet", group: :development
+gem "sorbet-runtime"
```
[the remark this answers is not in the document]
Comment (tsmartt) [reply Gemfile:140]: Done.

**Comment by tsmartt** [comment]
> Can we exclude rbi from the changes? They seems autogenerated and we should not edit them anyway
They are [meant to be vendored](https://sorbet.org/docs/rbi#a-note-about-vendoring-rbis) for explicit upgrade reasons

> Can we split all the # type: ignore changes in a separated commit?
This PR will get squashed into 1 commit when merged.

**Comment by thypon** [comment]
GTG

**Pull request merged**